nomad/deployment_endpoint.go

```diff
@@ -450,20 +450,33 @@ func (d *Deployment) List(args *structs.DeploymentListRequest, reply *structs.De
 
 	// Check namespace read-job permissions against request namespace since
 	// results are filtered by request namespace.
-	if aclObj, err := d.srv.ResolveACL(args); err != nil {
+	aclObj, err := d.srv.ResolveACL(args)
+	if err != nil {
 		return err
-	} else if aclObj != nil && !aclObj.AllowNsOp(namespace, acl.NamespaceCapabilityReadJob) {
+	}
+	
+	if aclObj != nil && !aclObj.AllowNsOp(namespace, acl.NamespaceCapabilityReadJob) {
 		return structs.ErrPermissionDenied
 	}
 
+	allow := aclObj.AllowNsOpFunc(acl.NamespaceCapabilityReadJob)
+
 	// Setup the blocking query
 	sort := state.SortOption(args.Reverse)
 	opts := blockingOptions{
 		queryOpts: &args.QueryOptions,
 		queryMeta: &reply.QueryMeta,
 		run: func(ws memdb.WatchSet, store *state.StateStore) error {
+			allowableNamespaces, err := allowedNSes(aclObj, store, allow)
+			if err != nil {
+				if err == structs.ErrPermissionDenied {
+					reply.Deployments = make([]*structs.Deployment, 0)
+					return nil
+				}
+				return err
+			}
```

Comment:
Not blocking but something we might want to get agreement on with the team and then document in our RPC checklist...

There's a small inconsistency here between the behavior of the first permissions check (on line 458) and this permissions check in the blocking query:
* If you have no permission to any namespace when you make the query, you get `ErrPermissionDenied`
* If you have permissions and make a blocking query, and while you're waiting on the blocking query the permissions are removed, you get an empty list.

I spot-checked some of the other `List` RPCs and it looks like we're inconsistent with this across the code base; some RPCs have both checks (ex. `Eval.List`, `Job.List`) while others only have the second check (ex. `Namespaces.ListNamespace`, `Variables.List`).